Bridges/HttpKernel.php

```diff
@@ -191,23 +181,19 @@ protected function mapResponse(HttpResponse $reactResponse, SymfonyResponse $syR
                 $name = substr($header, 0, $pos);
                 $value = trim(substr($header, $pos + 1));
 
-                if (isset($nativeHeaders[$name])) {
-                    if (!is_array($nativeHeaders[$name])) {
-                        $nativeHeaders[$name] = [$nativeHeaders[$name]];
-                    }
-
-                    $nativeHeaders[$name][] = $value;
-                } else {
-                    $nativeHeaders[$name] = $value;
-                }
+                $nativeHeaders[$name][] = $value;
             }
         }
 
         // after reading all headers we need to reset it, so next request
         // operates on a clean header.
         header_remove();
 
-        $headers = array_merge($nativeHeaders, $syResponse->headers->allPreserveCase());
+        $headers = array_merge(
```

Comment:
lowering the keys might be a bc break ?

Reply:
Where exactly do you see the BC break? It's just outputting of the headers, and Aerys lowercases them anyway.

Reply:
After a more in depth look I couldnt find a actual problem